python/cudf/cudf/core/column/struct.py

```diff
@@ -191,7 +191,15 @@ def field(self, key):
             pos = fields.index(key)
             return self._return_or_inplace(self._column.children[pos])
         else:
-            return self._return_or_inplace(self._column.children[key])
+            try:
+                return self._return_or_inplace(self._column.children[key])
+            except Exception:
+                print(
+                    'Field "'
+                    + str(key)
+                    + '" is not found in the set of existing keys.'
+                )
+                raise KeyError
```

Comment:
A couple of comments here:

1. When we hit the `else` block in `193`, the assumption is basically that `key` is an integer of some sort. If it is not, we can go ahead and raise the `KeyError` immediately.

2. Once that logic in place, we can catch a more specific exception than just `Exception`. Specifically, `children[key]` will raise an `IndexError` if `key >= num_children`.

3. Finally, we should prefer `raise KeyError(error_message_string)` v/s `print(error_message_string); raise KeyError`

Reply:
Is it possible we end up in this code path with an empty struct column? If so, we might still hit a situation where we call do a getitem[0] on an empty tuple

Reply:
It's possible, but even in that case, we will catch the resulting `IndexError` and re-raise as a `KeyError`, which I think is the right behaviour.

Reply:
Ah, right. This is implicit in your steps above. Makes sense!

Reply:
Thanks for the comments, @shwina and @beckernick!!  I'll implement your suggested changes!

Reply:
I believe this conversation can now be resolved;  let me know what you think of the new implementation!

Reply:
Looks great -- resolving!